assets/settings/default.json

```diff
@@ -297,6 +297,8 @@
     // Whether to show warnings or not by default.
     "include_warnings": true
   },
+  // custom file associations
```

Comment:
Style nit: all other settings start with the uppercase letter and end with the dot, let's keep the new entries the same.

Severe style issue: default settings is currently one of the main sources of the "documentation" for Zed users.
It should be obvious how to configure the file associations by looking at those, so let's add a comment with the example above.
And we also would need an entry here: https://github.com/zed-industries/zed/blob/main/docs/src/configuring_zed.md